### ISSUE TITLE
CMake 3.11.0 breaks certain vcpkg builds

### ISSUE THREAD

**Issue by past-due** [appveyor/ci#2231]
See: https://github.com/Microsoft/vcpkg/issues/2985

With the recent build environment update (#2180) to CMake 3.11.0, certain vcpkg builds have broken. (Supposedly due to a bug in CMake 3.11.)

**Comment by IlyaFinkelshteyn** [comment]
We need a repro, preferable hello-world-style to test before next image update.

**Comment by past-due** [comment]
This should do it (at least until vcpkg themselves pushes a workaround / fix):

```
git clone -q https://github.com/Microsoft/vcpkg.git;
pushd vcpkg;
.\bootstrap-vcpkg.bat;
.\vcpkg install openal-soft
popd;
```

**Comment by past-due** [comment]
As an interim workaround for anyone affected, add the following lines to the appropriate section of your appveyor.yml to disable the pre-installed CMake in the build environment:

```
- ps: if (Test-Path "${env:ProgramFiles}\CMake") { Rename-Item -Path "${env:ProgramFiles}\CMake" -NewName "CMake_disabled" -ErrorAction SilentlyContinue; Write-Output "Disabled CMake at ${env:ProgramFiles}\CMake"; }
- ps: if (${env:ProgramFiles(x86)} -ne $null) { if (Test-Path "${env:ProgramFiles(x86)}\CMake") { Rename-Item -Path "${env:ProgramFiles(x86)}\CMake" -NewName "CMake_disabled" -ErrorAction SilentlyContinue; Write-Output "Disabled CMake (32-bit) at ${env:ProgramFiles(x86)}\CMake"; } }
```

**Comment by past-due** [comment]
@FeodorFitsner: Yep - the workaround above renames the pre-installed CMake in the AppVeyor image to trigger vcpkg to fallback to downloading a portable CMake 3.10.2 for building dependencies (which works).

My understanding is that CMake 3.11.0 contains changes (bugs?) that break the ability to build various dependencies with vcpkg. https://github.com/Microsoft/vcpkg/issues/2985 says they will eventually have a blacklist of broken CMake versions, which should handle falling back to a supported portable CMake when necessary, but I don't see any evidence that this is ready yet.

**Comment by FeodorFitsner** [comment]
Looking at https://github.com/Microsoft/vcpkg/issues/2985 - I can understand "rc" release can be "bad", but 3.11.0 is RTM'ed now. I'd communicate the issue to CMake team anyway, but I'm not in a right position to do that as I'm not an expert in C++ stuff :)

**Comment by ras0219-msft** [comment]
This _particular_ issue looks like a bug in `openal-soft` -- I'm investigating a fix.

**Comment by Naarii528** [comment]
Yup, I was about to open an issue myself, but then I saw one that seemed to have solved the issue, by using 3.10 of Cmake. But, I am now on that version, and I am still getting an error when trying to install openal-soft. If this is worth making a new issue, let me know.

**Comment by ras0219-msft** [comment]
I've checked in a fix to vcpkg's master branch for `Openal-soft` in cmake 3.11. Please let us know if you have any more issues with vcpkg and cmake 3.11!

**Comment by past-due** [comment]
I can conform that https://github.com/Microsoft/vcpkg/commit/3baf583934f3077070e9ed4e7684f743ecced577 fixes the error building `openal-soft` with CMake 3.11.

@FeodorFitsner: So perhaps it makes sense to update the pre-installed vcpkg in the build environment to that commit (or newer)?

**Comment by FeodorFitsner** [comment]
Sure, will update this next week. Thanks!

**Comment by JayFoxRox** [comment]
A small update on this: kcat, the openal-soft maintainer has fixed this as an issue in openal-soft itself, as previously speculated (?) by vcpkg-maintainer ras0219-msft.
[A user then asked for an updated release on the respective commit, however, this will only happen in "probably in a month or two" ](https://github.com/kcat/openal-soft/commit/cae4b1a062b53dd25eba7caa41622be730106749)

The vcpkg still seems to be broken in the "Visual Studio 2017" image. The update (or patch) of vcpkg (which fixed this issue as an openal-soft 1.82 patch) did not happen last week ( @FeodorFitsner ).

I'd like to see this fixed soon so my builds work fine again. I'd also prefer to avoid adding a temporary hack in my project for an issue which should only remain for a couple of days anyway (as vcpkg should have been updated by now in the images).

**Comment by FeodorFitsner** [comment]
Should be fixed with these updates:
https://github.com/appveyor/ci/issues/2287
https://github.com/appveyor/ci/issues/2281
